File: examples/src/Tables/GetEffectiveValuesOfTable.cs
﻿import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.Tables
{
    class GetEffectiveValuesOfTable
    {
        public static void Run() {

            #ExStart:GetEffectiveValuesOfTable

            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Tables()

            using (Presentation pres = new Presentation(dataDir + "pres.pptx"))
            {
                ITable tbl = pres.slides[0].shapes[0] as ITable
                ITableFormatEffectiveData tableFormatEffective = tbl.TableFormat.GetEffective()
                IRowFormatEffectiveData rowFormatEffective = tbl.Rows[0].RowFormat.GetEffective()
                IColumnFormatEffectiveData columnFormatEffective = tbl.Columns[0].ColumnFormat.GetEffective()
                ICellFormatEffectiveData cellFormatEffective = tbl[0, 0].CellFormat.GetEffective()

                IFillFormatEffectiveData tableFillFormatEffective = tableFormatEffective.FillFormat
                IFillFormatEffectiveData rowFillFormatEffective = rowFormatEffective.FillFormat
                IFillFormatEffectiveData columnFillFormatEffective = columnFormatEffective.FillFormat
                IFillFormatEffectiveData cellFillFormatEffective = cellFormatEffective.FillFormat
               
            }
            #ExEnd:GetEffectiveValuesOfTable

        }
    }
}

File: examples/src/Tables/get_effective_values_of_table.py
import aspose.slides as slides


def get_effective_values_of_table(global_opts):
    with slides.Presentation(global_opts.data_dir + "tables.pptx") as pres:
        table = pres.slides[0].shapes[0]
        table_format_effective = table.table_format.get_effective()
        row_format_effective = table.rows[0].row_format.get_effective()
        column_format_effective = table.columns[0].column_format.get_effective()
        cell_format_effective = table.rows[0][0].cell_format.get_effective()

        table_fill_format_effective = table_format_effective.fill_format
        row_fill_format_effective = row_format_effective.fill_format
        column_fill_format_effective = column_format_effective.fill_format
        cell_fill_format_effective = cell_format_effective.fill_format
